app/react/ToggledFeatures/tocGeneration/TocGeneratedLabel.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { FeatureToggle } from 'app/components/Elements/FeatureToggle';
+import { FileType } from 'shared/types/fileType';
+
+export interface TocGeneratedLabelProps {
+  file: FileType;
+  children: JSX.Element | string;
+}
+
+export const TocGeneratedLabel = ({ file, children }: TocGeneratedLabelProps) => (
+  <FeatureToggle feature="tocGeneration">
+    {file.generatedToc && <span className="badge">{children}</span>}
```

Comment:
If children can be a JSX.Element, `<span>` only allows a very few children. Should this may be an inline div? Should it just accept strings?

Reply:
this was to allow <Translate> or a string, since it can not be specified [which element is allowed as a TS type](https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase#what-you-cannot-do)
i have changed it for a div as you suggested, badge is "inline-block"